File: src/main/java/dev/lbuddyboy/flash/user/comparator/GrantDateComparator.java
package dev.lbuddyboy.flash.user.comparator;

import dev.lbuddyboy.flash.user.model.Grant;

import java.util.Comparator;

public class GrantDateComparator implements Comparator<Grant> {

    @Override
    public int compare(Grant grant, Grant otherGrant) {
        return Long.compare(grant.getAddedAt(), otherGrant.getAddedAt());
    }
}

File: src/main/java/dev/lbuddyboy/flash/user/comparator/GrantDateComparator.kt
package dev.lbuddyboy.flash.user.comparator

import dev.lbuddyboy.flash.user.model.Grant

class GrantDateComparator : Comparator<Grant> {
    override fun compare(grant: Grant, otherGrant: Grant): Int {
        return java.lang.Long.compare(grant.addedAt, otherGrant.addedAt)
    }
}